src/datatypes/src/value.rs

```diff
@@ -827,8 +827,10 @@ impl TryFrom<ScalarValue> for Value {
             ScalarValue::DurationNanosecond(d) => d
                 .map(|x| Value::Duration(Duration::new(x, TimeUnit::Nanosecond)))
                 .unwrap_or(Value::Null),
-            ScalarValue::Decimal128(_, _, _)
-            | ScalarValue::Decimal256(_, _, _)
+            ScalarValue::Decimal128(v, p, s) => v
+                .map(|v| Value::Decimal128(Decimal128::new(v, p, s)))
```

Comment:
How about using `try_new()` instead

Reply:
the scalarvalue from datafusion can promise the safety